### ISSUE TITLE
Site Search - API - Not considering "Category"

### ISSUE THREAD

**Issue by siva538** [matomo-org/matomo#13459]
Hi Matomo team,

Thanks for the wonderful product. While the Search tracking considers "Category" for any of the captures in the application, the same cannot be reported/retrieved back using the API.

So looking for something like:

 Actions.getSiteSearchKeywordsByCategory (idSite, period, date, category, segment = '')

instead of just 

 Actions.getSiteSearchKeywords (idSite, period, date, segment = '')

This actually makes sense to me because without Category, the keywords data is meaningless when I am searching in two different entities altogether like ex: a multi-module seller might sell Cars as well as bikes. Searches relevant to cars retrieval is needed for consuming the contents of what all the users are looking for, instead of a combined list of searches for both cars and bikes.

You may consider this as an enhancement request or bug. But would be helpful if you can consider this request.

Thanks in advance. 

Kumar

**Comment by tsteur** [comment]
Makes sense. Maybe we instead rather add support for a `itemCategory` segment @mattab ? I'll change the title for now... 

The segment would automatically search in all of the 5 categories.

Not sure how easy it is to add such a segment... I presume when there are many items per order, and the archiver does a left join, there might be a group by and a subquery missing or so. To be seen.

**Comment by siva538** [comment]
Sorry for interrupting on your discussion. I do not understand the reasoning behind introduction of "Segments" into this request.

The ask is, the way we are storing the tracking contents based on a category, during the time of tracking of Site Search, exactly the same way retrieval should support the "Category" based.

So requesting for enhancing of API to include the Category as well, without which the Keywords data doesn't mean anything in this - 

Actions.getSiteSearchKeywords (idSite, period, date, segment = '')

@tsteur , @mattab - Please let me know if my understanding is incorrect.

**Comment by tsteur** [comment]
Using a segment you will be able to get this information. And you will be able to get even any other reports segmented by category.

**Comment by siva538** [comment]
Hi @tsteur . We have validated this with Segmentation today but no luck. Data is not coming up as expected in the "Site Search" reports. Please suggest what else I should be doing.

**Comment by siva538** [comment]
Hello @tsteur / @mattab , while the initial request on this was to have the provision for "Category" being placed in the Reporting API, it turned out to be the UI dashboards are also not supporting category based. 

Best example could be the "Event Category" field provided as a dimension data in the report for when viewing the "Events". 

The same functionality is not available in "Site Search" which is making us to use ONLY 1 category in the entire application, and hence only 1 type of "Search" could be tracked. 

The segmentation also will not work based on the site category as it was not provided as a filtering field, and assuming it is provided in the future, it is not the optimal way of doing it.

Request you to consider the provision of Search Category, in the Site Search subsection, so we could track multiple types of Searches and not limiting ourselves to only 1 type.

Regds,
Sivakumar

**Comment by mattab** [comment]
Hi Sivakumar, do you mind showing a screenshot of the Segment you created, and a screenshot showing the no data screen?

**Comment by siva538** [comment]
Apologies for the delay in responding to this @mattab . There is no screenshot I could share as I mentioned above that the field is not available for filtering the segments for "Search Category".
